src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx

```diff
@@ -0,0 +1,149 @@
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as UserUtils from '@libs/UserUtils';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import type {WithPolicyAndFullscreenLoadingProps} from '@pages/workspace/withPolicyAndFullscreenLoading';
+import withPolicyAndFullscreenLoading from '@pages/workspace/withPolicyAndFullscreenLoading';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Route} from '@src/ROUTES';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {PersonalDetails, PersonalDetailsList} from '@src/types/onyx';
+
+type WorkspacePolicyOnyxProps = {
+    /** Personal details of all users */
+    personalDetails: OnyxEntry<PersonalDetailsList>;
+};
+
+type WorkspaceMemberDetailsPageProps = WithPolicyAndFullscreenLoadingProps & WorkspacePolicyOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.MEMBER_DETAILS>;
+
+function WorkspaceMemberDetailsPage({personalDetails, policyMembers, policy, route}: WorkspaceMemberDetailsPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const StyleUtils = useStyleUtils();
+
+    const [removeMemberConfirmModalVisible, setRemoveMemberConfirmModalVisible] = React.useState(false);
+
+    const accountID = Number(route?.params?.accountID) ?? 0;
+    const backTo = decodeURIComponent(route?.params?.backTo ?? '') as Route;
+
+    const member = policyMembers?.[accountID];
+    const details = personalDetails?.[accountID] ?? ({} as PersonalDetails);
+    const avatar = details.avatar ?? UserUtils.getDefaultAvatar();
+    const fallbackIcon = details.fallbackIcon ?? '';
+    const displayName = details.displayName ?? '';
+
+    const askForConfirmationToRemove = () => {
+        setRemoveMemberConfirmModalVisible(true);
+    };
+
+    const removeUser = useCallback(() => {
+        Policy.removeMembers([accountID], route.params.policyID);
+        setRemoveMemberConfirmModalVisible(false);
+        Navigation.goBack(backTo);
+    }, [accountID, backTo, route.params.policyID]);
+
+    const redirectToProfile = useCallback(() => {
+        Navigation.navigate(ROUTES.PROFILE.getRoute(accountID));
+    }, [accountID]);
+
+    const openRoleSelectionModal = useCallback(() => {
+        const activeRoute = encodeURIComponent(Navigation.getActiveRouteWithoutParams());
+        Navigation.navigate(ROUTES.WORKSPACE_MEMBER_ROLE_SELECTION.getRoute(route.params.policyID, accountID, activeRoute));
+    }, [accountID, route.params.policyID]);
+
+    return (
+        <ScreenWrapper testID={WorkspaceMemberDetailsPage.displayName}>
+            <FullPageNotFoundView>
```

Comment:
added, but now I get `Page not found` page because my Workspaces have `type: 'free'` - is there a way to create "team" or "corporate" Workspaces?

Reply:
ok, I think disabling RHN for non Paid Workspace is most reasonable - I've made this change, so now:

- for Paid Workspaces, when user presses the checkbox, the item is selected and when user presses the whole item it shows the Member Details modal
- for Non Paid Workspaces, pressing the whole item, as well as pressing the checkbox just selects the item

Please see the recording:

https://github.com/Expensify/App/assets/7771498/e9cd5d8c-2d83-424b-92be-059de2044f1c

What do you think?

Reply:
> for Non Paid Workspaces, pressing the whole item, as well as pressing the checkbox just selects the item

Personally I don't like that there would be variability in pressing the whole item triggering the select action or not. I think we should limit selecting a row to just engaging with the checkbox, no matter what the case is. Curious if @Expensify/design agrees here. 

So basically, let's not try to get tricky with using the whole row as a trigger action for the select box. Just make it so that you can only ever use the select box to select a row. For non-paid workspaces, just make the rest of the row non-tappable.

Reply:
@shawnborton makes sense to me - I'll change it

Reply:
Totally agree @shawnborton.